src/StoreApi/Routes/V1/OrderUpdateCustomer.php

```diff
@@ -0,0 +1,135 @@
+<?php
+namespace Automattic\WooCommerce\StoreApi\Routes\V1;
+
+use Automattic\WooCommerce\StoreApi\SchemaController;
+use Automattic\WooCommerce\StoreApi\Schemas\v1\AbstractSchema;
+use Automattic\WooCommerce\StoreApi\Utilities\OrderAuthorizationTrait;
+use Automattic\WooCommerce\StoreApi\Utilities\OrderController;
+use Automattic\WooCommerce\StoreApi\Utilities\UpdateCustomerTrait;
+
+/**
+ * OrderUpdateCustomer class.
+ *
+ * Updates the customer billing and shipping addresses and returns an updated order--taxes should be recalculated.
+ */
+class OrderUpdateCustomer extends AbstractRoute {
+	use OrderAuthorizationTrait;
+	use UpdateCustomerTrait {
+		get_args as protected get_update_customer_args;
+	}
+
+	/**
+	 * The route identifier.
+	 *
+	 * @var string
+	 */
+	const IDENTIFIER = 'order-update-customer';
+
+	/**
+	 * The schema item identifier.
+	 *
+	 * @var string
+	 */
+	const SCHEMA_TYPE = 'order';
+
+	/**
+	 * Order controller class instance.
+	 *
+	 * @var OrderController
+	 */
+	protected $order_controller;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param SchemaController $schema_controller Schema Controller instance.
+	 * @param AbstractSchema   $schema Schema class for this route.
+	 */
+	public function __construct( SchemaController $schema_controller, AbstractSchema $schema ) {
+		parent::__construct( $schema_controller, $schema );
+
+		$this->order_controller = new OrderController();
+	}
+
+	/**
+	 * Get the path of this REST route.
+	 *
+	 * @return string
+	 */
+	public function get_path() {
+		return '/order/(?P<id>[\d]+)/update-customer';
+	}
+
+	/**
+	 * Get method arguments for this REST route.
+	 *
+	 * @return array An array of endpoints.
+	 */
+	public function get_args() {
+		$args = $this->get_update_customer_args();
+
+		foreach ( $args as $index => $arg ) {
+			$args[ $index ]['permission_callback'] = [ $this, 'is_authorized' ];
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Handle the request and return a valid response for this endpoint.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_REST_Response
+	 */
+	protected function get_route_post_response( \WP_REST_Request $request ) {
+		$order_id = absint( $request['id'] );
+		$order    = wc_get_order( $order_id );
+
+		// Get data from request object and merge with order customer data, then sanitize.
+		$billing  = $this->schema->billing_address_schema->sanitize_callback(
```

Comment:
Sorry to be pedantic but can we call these `$billing_address` and `$shipping_address` just to make it a little easier to reason with?

Reply:
Sure, we can update these.  For reference, these were following [existing patterns in the card endpoint](https://github.com/woocommerce/woocommerce-blocks/blob/469b7304715341d30d75b7d5b6629f089c08399d/src/StoreApi/Routes/V1/CartUpdateCustomer.php#L43).

Reply:
Ah, if the pattern already exists lets keep it as it is. No point diverging.